jobs/scheduler/templates/pre-start.erb

```diff
@@ -33,22 +33,40 @@ USER='<%= role['name'] %>'
 PASSWORD='<%= role['password'] %>'
 PORT='<%= p("autoscaler.scheduler_db.port") %>'
 SSLMODE='<%= p("autoscaler.scheduler_db.sslmode") %>'
+<% if p('autoscaler.scheduler_db.db_scheme') == "postgres" %>
 JDBCDBURL="jdbc:postgresql://$HOST:$PORT/$DBNAME?sslmode=$SSLMODE"
 DBURI="postgres://$USER:$PASSWORD@$HOST:$PORT/$DBNAME?sslmode=$SSLMODE"
-
-<% unless p('autoscaler.scheduler_db.tls.ca') == "" %>
+<% unless p('autoscaler.scheduler_db.tls.ca') == "" %> 
 JDBCDBURL=$JDBCDBURL"&sslrootcert=/var/vcap/jobs/scheduler/config/certs/scheduler_db/ca.crt"
 DBURI=$DBURI"&sslrootcert=/var/vcap/jobs/scheduler/config/certs/scheduler_db/ca.crt"
 <% end %>
+<% else %>
+JDBCDBURL="jdbc:mysql://$HOST:$PORT/$DBNAME?autoReconnect=true"
+DBURI="$USER:$PASSWORD@tcp($HOST:$PORT)/$DBNAME?tls=$SSLMODE"
+<% unless p('autoscaler.scheduler_db.tls.ca') == "" %>
+/var/vcap/jobs/scheduler/bin/install_crt_truststore scheduler_db  scheduler_db/ca.crt
+JDBCDBURL=$JDBCDBURL"&useSSL=true&requireSSL=true&verifyServerCertificate=true&trustCertificateKeyStorePassword=123456&trustCertificateKeyStoreUrl=file:/var/vcap/data/certs/scheduler_db/cacerts&trustCertificateKeyStoreType=pkcs12"
+DBURI=$DBURI"&sslrootcert=/var/vcap/jobs/scheduler/config/certs/scheduler_db/ca.crt"
+<% end %>
+<% end %>
 
 detect_dns $HOST $PORT
 /var/vcap/packages/changeloglockcleaner/changeloglockcleaner -d "$DBURI" -e <%= p("autoscaler.changeloglock_timeout_seconds") %>
+<% if p('autoscaler.scheduler_db.db_scheme') == "postgres" %>
 java -cp "$DB_DIR/target/lib/*" liquibase.integration.commandline.Main --url "$JDBCDBURL" --username=$USER --password=$PASSWORD \
 --driver=org.postgresql.Driver --changeLogFile=$SCHEDULER_DIR/db/scheduler.changelog-master.yaml update
 
 java -cp "$DB_DIR/target/lib/*" liquibase.integration.commandline.Main --url "$JDBCDBURL" --username=$USER --password=$PASSWORD \
 --driver=org.postgresql.Driver --changeLogFile=$SCHEDULER_DIR/db/quartz.changelog-master.yaml update
+<% else %>
+java -cp "/var/vcap/packages/db/target/lib/*" liquibase.integration.commandline.Main --url "$JDBCDBURL" --username=$USER --password=$PASSWORD \
+--driver=com.mysql.cj.jdbc.Driver --changeLogFile=$SCHEDULER_DIR/db/databasechangelog.yaml update
```

Comment:
why postgres don't need this?

Reply:
When using cf mysql pxc cluster, it requires primary key for every table. So I added a changelog here to add thee primary key for "DATABASECHANGELOG". 
postgres do not require the primary key.